draft-ietf-tls-tls13.md

```diff
@@ -5450,6 +5505,32 @@ ClientHello without 0-RTT data.
 To avoid this error condition, multi-server deployments SHOULD ensure a uniform
 and stable deployment of TLS 1.3 without 0-RTT prior to enabling 0-RTT.
 
+## Middlebox Compatibility Mode {#middlebox}
+
+Field measurements have found that a significant number of middleboxes
+misbehave when a TLS client/server pair negotiates TLS 1.3. Implementations
+can increase the chance of making connections through those middleboxes
+by making the TLS 1.3 handshake look more like a TLS 1.2 handshake:
+
+- The client always provides a non-empty session ID in the ClientHello.
+
+- If not offering early data, the client sends a dummy
```

Comment:
what does "dummy" mean here? It's a term not used anywhere else in the draft.

Does the CCS has to be always sent? If not, what the other side is supposed to do when it doesn't receive it? What if it receives it at different time? What if it receives multiple CCS messages one after another?

Reply:
The receiver ignores all CCS records during the handshake, without any care as to where they do or don't appear. It's in the record layer section.

This is based on implementor feedback. Our initial prototype had the receiver enforce the CCS positions, but the handshake/CCS synchronization on the read side is a bit of a mess in TLS 1.2. There was a desire from some folks to not complicate their TLS 1.3 state machined with this. Discarding them at the record layer, on the other hand, is straightforward. (Additionally, it allows any implementation to unilaterally disable compatibility mode if it is deployed somewhere these middlebox issues are less common.)

Reply:
Perhaps specify that a "dummy CCS record" is one consisting of a single `0x01` byte.

Reply:
I meant that "dummy" is obvious to us, in this discussion, in the context of this PR. Not necessarily to somebody reading the spec in 5 years.

Reply:
I don't think dummy is confusing here.